definitions/checks/check_for_newer_packages.rb

```diff
@@ -12,15 +12,54 @@ class Checks::CheckForNewerPackages < ForemanMaintain::Check
   end
 
   def run
-    exit_status, = ForemanMaintain.package_manager.check_update(packages: @packages,
-                                                                with_status: true)
-    assert(exit_status == 0, 'An update is available for one of these packages: '\
-                "#{@packages.join(',')}. Please update before proceeding.")
+    check_for_package_update
     if @manual_confirmation_version
       question = 'Confirm that you are running the latest minor release of Satellite '\
                 "#{@manual_confirmation_version}"
       answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
       abort! if answer != :yes
     end
   end
+
+  def check_for_package_update
+    exit_status, output = ForemanMaintain.package_manager.check_update(packages: @packages,
+                                                                       with_status: true)
+    packages_with_updates = []
+    if exit_status == 100
+      packages_with_updates = compare_pkg_versions(output).select do |_, result|
+        result == -1
+      end.keys
+    end
+
+    unless packages_with_updates.empty?
+      failure_message = 'An update is available for package(s): '\
+                        "#{packages_with_updates.join(',')}. Please update before proceeding!"
+      fail! failure_message
+    end
+  end
+
+  def packages_and_versions(output)
+    pkgs_versions = {}
+    pkg_details = output.split("\n\n")[1].split
+    iterator = 0
+    while iterator < pkg_details.length
+      break if pkg_details[iterator].nil?
+      pkg_name = pkg_details[iterator].split('.').first
+      pkg_version = version(pkg_details[iterator + 1].split('-').first)
+      pkgs_versions[pkg_name] = pkg_version
+      iterator += 3
+    end
```

Comment:
```suggestion
  @packages.each do |pkg|
    output.lines.each do |line|
      line.strip!
      next unless line.start_with?(pkg)
      pkg_name, pkg_version, repo = line.split
      pkgs_versions[pkg_name] = version(pkg_version.split('-').first)
    end
  end
  
  how about this
```

Reply:
Thats better however looping with `.lines` does not work as its splitting just after the package `python3-pulp-2to3-migration`,
~~~
"Loaded plugins: product-id, search-disabled-repos, subscription-manager\n\npython3-pulp-2to3-migration.noarch\n                             0.11.4-1.el7pc     rhel-7-server-satellite-6.9-rpms\ntfm-rubygem-katello.noarch   3.18.1.40-1.el7sat rhel-7-server-satellite-6.9-rpms"
~~~
The same logic with array works, pushing this change now!
~~~
    @packages.each do |pkg|
      pkg_details.each_with_index do |ele, index|
        next unless ele.start_with?(pkg)
        pkgs_versions[pkg] = version(pkg_details[index+1].split('-').first)
        break
      end
    end
~~~

Reply:
Eww, yeah, yum does this sometimes with too long lines :/

Thanks!